src/remote.rs

```diff
@@ -50,6 +50,14 @@ pub struct PushOptions<'cb> {
     pb_parallelism: u32,
 }
 
+/// Holds callbacks for a connection to a `Remote`. Disconnects when dropped
+#[allow(dead_code)]
```

Comment:
The `callbacks` and `proxy` fields are assigned to hold the callbacks but never used.

```
warning: field is never used: `callbacks`, #[warn(dead_code)] on by default
  --> src/remote.rs:55:5
   |
55 |     callbacks: Box<RemoteCallbacks<'cb>>,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: field is never used: `proxy`, #[warn(dead_code)] on by default
  --> src/remote.rs:56:5
   |
56 |     proxy: Box<ProxyOptions<'cb>>,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```